skipchain/struct.go

```diff
@@ -485,6 +485,58 @@ func (sb *SkipBlock) updateHash() SkipBlockID {
 	return sb.Hash
 }
 
+// Proof is a list of blocks from the genesis to the latest block
+// using the shortest path
+type Proof []*SkipBlock
+
+// Verify checks that the proof is correct by checking individual
+// blocks and their back and forward links
+func (sbs Proof) Verify() error {
+	if len(sbs) == 0 {
+		return errors.New("Empty list of blocks")
+	}
+
+	if sbs[0].Index != 0 {
+		return errors.New("First element must be a genesis")
+	}
+
+	for i, sb := range sbs {
+		if !sb.CalculateHash().Equal(sb.Hash) {
+			return errors.New("Wrong hash")
+		}
+		if i > 0 {
+			// Check if there is a back link to the previous block
+			hit := false
+			for _, bl := range sb.BackLinkIDs {
+				if bl.Equal(sbs[i-1].Hash) {
+					hit = true
+				}
+			}
+
+			if !hit {
+				return fmt.Errorf("Wrong backlink %v %v", sb.BackLinkIDs, sbs[i-1].Hash)
+			}
+		}
+		if i < len(sbs)-1 {
+			// Check if there is a forward link to the next block
+			if len(sb.ForwardLink) == 0 {
+				return errors.New("Missing forward links")
+			}
+
+			fl := sb.ForwardLink[len(sb.ForwardLink)-1]
```

Comment:
are you not checking all the forward links?

Reply:
Only the highest one because the block for the lower ones are not propagated